next.clayui.com/content/docs/foundations/composing.md

```diff
@@ -3,4 +3,146 @@ title: "Composing"
+
+</div>
+</div>
+
+Clay follows some fundamentals for creating new components and as we distribute the components so that they can be used, the components follow some of the main React patterns: [composition](https://reactjs.org/docs/composition-vs-inheritance.html) or you can listen around for "bulding blocks".
+
+We believe that using composition is the most correct and is the most widely used standard in the community. We want to allow people to reuse our components for different purposes, we understand that we have different people with different opinions on how we want to use Clay for their purposes. We think about how we can have a little of both worlds.
+
+## Composition
+
+The composition is very strong with React, JSX syntactic sugar makes it very powerful and more viable when coupled with special `children` prop for passing arbitrary child elements. Props in React components allows us to pass objects and consequently React elements since they are objects.
+
+```jsx
+<ClayModal>
+	<ClayModal.Footer
+		first={
+			<ClayButton.Group spaced>
+				<ClayButton displayType="secondary">
+					{'Secondary'}
+				</ClayButton>
+				<ClayButton displayType="secondary">
+					{'Secondary'}
+				</ClayButton>
+			</ClayButton.Group>
+		}
+		middle={
+			<ClayButton>{'Primary'}</ClayButton>
+		}
+	/>
+</ClayModal>
+```
+
+So we use some of the composable component approach in React.
+
+<div class="clay-ullist">
+
+- [**Containment**](https://reactjs.org/docs/composition-vs-inheritance.html#containment) - They are treated as generic blocks that accept the passing of arbitrary `children` to them, either using `children` props or the special naming convention itself.
+- [**Specialization**](https://reactjs.org/docs/composition-vs-inheritance.html#specialization) - "Special case" components that are specific for some use purposes and not treated as something more "generic".
+
+</div>
+
+## Philosophy
+
+Assuming that we want our components to be reusable and enable developers to have the flexibility to build on their use cases while helping developers who want to do something fast when just following [Lexicon](https://lexicondesign.io) specifications, we follow a few Thought lines.
+
+Teams are constantly advancing and new features are always emerging that require new interactions and behavioral changes, a product changes a lot over time, being refined, aligning interests and changing behaviors in the UI.
+
+Designers cannot get caught up in strict rules this blocks the creative mind and can prevent UI advances and interaction enhancements. The same goes for components, it has to allow developers to move forward with the standards made by designers and allow them to reuse components in the future.
+
+Dealing with this is complex and requires balancing component flexibility and specialization, getting to that balance is difficult and we can spend a lot of time polishing and refining components to achieve **"component supremacy"**.
```

Comment:
What even is "component supremacy"? Might be better to just leave that portion off.

```suggestion
Dealing with this is complex and requires balancing component flexibility and specialization, getting to that balance is difficult and we can spend a lot of time polishing and refining components.
```

Reply:
It's just a term to symbolize how difficult it would be to get the perfect balance of the component with regard to flexibility and specialization... Does that make sense to you?

Reply:
It kind of does, but I think it might be more confusing to have it in than if we just remove that portion. I think we still convey the same thing by leaving it out.

Reply:
Ok, I think this is enough.